mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSModelAsNode.kt

```diff
@@ -67,52 +86,73 @@ data class MPSModelAsNode(val model: SModel) : IDeprecatedNodeDefaults {
     override fun getChildren(link: IChildLink): Iterable<INode> {
         return if (link is NullChildLink) {
             emptyList()
-        } else if (link.getUID().endsWith(BuiltinLanguages.MPSRepositoryConcepts.Model.rootNodes.getUID()) ||
-            link.getUID().contains("rootNodes") ||
-            link.getSimpleName() == "rootNodes"
-        ) {
+        } else if (link.conformsTo(builtinModel.rootNodes)) {
             model.rootNodes.map { MPSNode(it) }
+        } else if (link.conformsTo(builtinModel.modelImports)) {
+            ModelImports(model).importedModels.mapNotNull {
+                MPSModelImportAsNode(it.resolve(model.repository), model)
+            }
+        } else if (link.conformsTo(builtinModel.usedLanguages)) {
+            getImportedLanguagesAndDevKits()
         } else {
             emptyList()
         }
     }
 
+    private fun getImportedLanguagesAndDevKits(): MutableList<INode> {
+        val importedLanguagesAndDevKits = mutableListOf<INode>()
+        if (model is SModelDescriptorStub) {
+            importedLanguagesAndDevKits.addAll(
+                model.importedLanguageIds().map {
+                    SingleLanguageDependencyAsNode(
+                        it.sourceModuleReference,
+                        model.getLanguageImportVersion(it),
+                        model,
+                    )
+                },
+            )
+            importedLanguagesAndDevKits.addAll(
+                model.importedDevkits().map { DevKitDependencyAsNode(it, model) },
+            )
+        }
+        return importedLanguagesAndDevKits
```

Comment:
I'd usually invert such an implementation in a way that preconditions are checked first with an early exist if not met. That way, the main body doesn't need another indentation level:

```suggestion
    private fun getImportedLanguagesAndDevKits(): MutableList<INode> {
        val smodel = model as? SModelDescriptorStub ?: return emptyList()
        
        val importedLanguagesAndDevKits = mutableListOf<INode>()
        importedLanguagesAndDevKits.addAll(
            smodel.importedLanguageIds().map {
                SingleLanguageDependencyAsNode(
                    it.sourceModuleReference,
                    smodel.getLanguageImportVersion(it),
                    smodel,
                )
            },
        )
        importedLanguagesAndDevKits.addAll(
            smodel.importedDevkits().map { DevKitDependencyAsNode(it, model) },
        )
        return importedLanguagesAndDevKits
```

The result list could also be created in an immutable fashion